motionplan/collision.go

```diff
@@ -242,10 +242,22 @@ func newCollisionGraph(x, y []spatial.Geometry, reference *collisionGraph, repor
 // checkCollision takes a pair of geometries and returns the distance between them.
 // If this number is less than the CollisionBuffer they can be considered to be in collision.
 func (cg *collisionGraph) checkCollision(x, y spatial.Geometry) (float64, error) {
+	// x is the robot geometries and therefore must be primitives in spatialmath
+	// y could potentially be a geometry type that lives outside spatialmath and therefore knows more so we defer to it for collisions
 	if cg.reportDistances {
-		return x.DistanceFrom(y)
+		dist, err := x.DistanceFrom(y)
+		if err != nil {
+			return y.DistanceFrom(x)
+		}
+		return dist, nil
 	}
 	col, err := x.CollidesWith(y)
+	if err != nil {
+		col, err = y.CollidesWith(x)
```

Comment:
[q] whats the difference in y colliding with x and x colliding with y?

Reply:
the Geometry which is having the CollidesWith function called on it matters. If CollidesWith is called on a sphere and an octree is passed in then there will be an error because sphere hasn't defined any collision checking with octrees. Thus we'll want to instead call CollidesWith on the octree and pass in a sphere because octree can check collisions with any Geometry type. It's a little hacky to get around the fact that the spatialmath Geometries can't check if they collide with an octree.

Reply:
If X doesn't know how to handle collisions with Y but Y knows how to handle the collision with X.  We try both now in case someone implemented a geometry that is not supported by our own primitives